packages/client/lib/net/protocol/ethprotocol.ts

```diff
@@ -117,7 +118,12 @@ export class EthProtocol extends Protocol {
       response: 0x04,
       encode: ({ reqId, block, max, skip = 0, reverse = false }: GetBlockHeadersOpts) => [
         bigIntToBuffer(reqId ?? ++id),
-        [typeof block === 'bigint' ? bigIntToBuffer(block) : block, max, skip, !reverse ? 0 : 1],
+        [
+          typeof block === 'bigint' ? bigIntToBuffer(block) : block,
+          max === 0 ? Buffer.from([]) : intToBuffer(max),
+          skip === 0 ? Buffer.from([]) : intToBuffer(skip),
+          !reverse ? Buffer.from([]) : Buffer.from([1]),
+        ],
```

Comment:
this is related to the RLP v3 changes, not sure why it just popped up in my PR, but when doing `bufArrToArr` we need all fields to be Buffer, otherwise if it is is a plain number it didn't convert correctly to Uint8Array.